x-pack/plugins/index_management/public/application/lib/discover_link.test.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { mountWithIntl } from '@kbn/test-jest-helpers';
+import { EuiButtonIcon } from '@elastic/eui';
+import { DiscoverLink } from './discover_link';
+import { AppContextProvider, AppDependencies } from '../app_context';
+
+describe('DiscoverLink', () => {
+  const indexName = 'my-fancy-index';
+
+  it('renders the link as an icon by default', async () => {
+    const navigateMock = jest.fn();
+    const ctx = {
+      url: {
+        locators: {
+          get: () => ({ navigate: navigateMock }),
+        },
+      },
+    } as unknown as AppDependencies;
+
+    const component = mountWithIntl(
+      <AppContextProvider value={ctx}>
+        <DiscoverLink indexName={indexName} />
+      </AppContextProvider>
+    );
+
+    expect(component.exists('[data-test-subj="discoverIconLink"]')).toBe(true);
+    expect(component.exists('[data-test-subj="discoverButtonLink"]')).toBe(false);
+  });
+
+  it('renders the link as a button if the prop is set', async () => {
```

Comment:
These tests are mostly from the renamed file `render_discover_link.test.ts` with an addition of new tests for rendering the link as a button (not only as an icon)